lib/ansible/modules/remote_management/manageiq/manageiq_dynamic_resource_definition.py

```diff
@@ -0,0 +1,283 @@
+
+        resource = {
+            'name': name,
+            'properties': properties,
+        }
+
+        try:
+            result = self.client.post(url, action='create', resource=resource)
+        except Exception as e:
+            self.module.fail_json(
+                msg="failed to create dynamic resource definition %s: %s" % (name, str(e))
+            )
+
+        return dict(
+            changed=True,
+            msg="successfully created the dynamic resource definition %s: %s" % (name, result['results'])
+        )
+
+    def has_field(self, dynamic_resource_definition, key):
+        """ Check if definition properties has a specific key.
+
+        If key is unhashble, return.
+        """
+        try:
+            if dynamic_resource_definition and dynamic_resource_definition.get('properties', {}).get(key):
+                return True
+        except TypeError:
+            return
+        else:
+            return
+
+    def edit_definition(self, dynamic_resource_definition, name, properties):
+        """ Edit the dynamic resource definition in manageiq.
+
+        Returns:
+            a dictionary with the changed and msg.
+        """
+        # Check if properties has the following keys,
+        # These keys are returned from the API response even if they are empty
+        # So we remove them to compare accuratley
+        for key in ['attributes', 'associations', 'methods']:
+            if not self.has_field(dynamic_resource_definition, key):
+                dynamic_resource_definition['properties'].pop(key)
+
+        if dynamic_resource_definition and name == dynamic_resource_definition.get(
+            'name'
+        ) and properties == dynamic_resource_definition.get(
+            'properties'
+        ):
+            return dict(
+                changed=False,
+                msg="Dynamic resource definition %s is not changed." % (name)
+            )
+
+        url = '%s/generic_object_definitions/%s' % (self.api_url, dynamic_resource_definition['id'])
+
+        resource = {
+            'name': name,
+            'properties': properties,
+        }
+
+        try:
+            result = self.client.post(url, action='edit', resource=resource)
+        except Exception as e:
+            self.module.fail_json(
+                msg="failed to edit dynamic resource definition %s: %s" % (name, str(e))
+            )
+
+        return dict(
+            changed=True,
+            msg="successfully edited the dynamic resource definition %s: %s" % (name, result)
+        )
+
+    def delete_definition(self, dynamic_resource_definition):
+        """ Delete the dynamic resource definition in manageiq.
+
+        Returns:
+            the name.
+        """
+        # check for required arguments
+        for key, value in dict(name=dynamic_resource_definition).items():
+            if value in (None, ''):
+                self.module.fail_json(msg="missing required argument: %s" % (key))
```

Comment:
I think this validation can be dropped too.  it's only `name` which is required=True.

Reply:
Sure, Will remove it.